### DISCUSSION TITLE
How to add custom attribute to <pre> element?

### DISCUSSION THREAD

**Discussion by amr3k** [pngwn/MDsveX#443] [Q&A] [answered]
I want to show a simple badge showing the programming language which a code was written into, something like this:

![screenshot](https://user-images.githubusercontent.com/12211826/161659448-35971026-033f-4b05-b061-58ccfb4368d7.png)

I tried using CSS only solution with pseudo element `::before` and used the content of `attrs(class)` but it prints the whole class name which is for example `language-php`.

So I'm thinking of injecting some code in mdsvex which adds something like `data-lang="php"` to the compiled md code.

Is it possible to do that?

Edit: Thinking about it again, I believe that I need to know how to be able to alter the rendered code. there's a situation where I want to add `target="_blank"` to all external links e.g. not starting with `/`.

**Comment by pngwn** [comment, marked as answer]
This is difficult at the minute due to this issue https://github.com/pngwn/MDsveX/issues/100 but will be resolved in 1.0 (which I'm working on atm).
